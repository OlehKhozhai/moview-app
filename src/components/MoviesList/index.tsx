import React from 'react';
import cn from 'classnames';

import MoviesListItem from 'components/MoviesListItem';
import { Movie } from 'redux/types';
import styles from './styles.module.scss';

type MoviesListProps = {
  movies: Movie[];
  isLoading: boolean;
  className?: string;
};

const MoviesList: React.FC<MoviesListProps> = ({ movies, isLoading, className }) => {
  return (
    <div className={cn(styles.root, className)}>
      {movies.length > 0 && (
        <>
          <h3 className={styles.counter}>{movies.length} movies found</h3>

          {movies.map((movie) => {
            return <MoviesListItem key={movie.id} {...movie} />;
          })}
        </>
      )}

      {movies.length === 0 && !isLoading && (
        <h2 className={styles.noMovieFound}>No movies found</h2>
      )}

      {isLoading && <h2>Loading</h2>}
    </div>
  );
};

export default React.memo(MoviesList);
