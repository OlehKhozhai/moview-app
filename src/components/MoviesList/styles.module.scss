@import 'assets/styles/mixins/colors';

.root {
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-flow: wrap;
  background-color: $primary-bg;
  color: $primary-color;
  padding: 20px 60px;
}

.counter {
  width: 100%;
  padding: 15px 0px 25px;
}

.noMovieFound {
  text-align: center;
  padding: 150px 0;
  width: 100%;
  font-size: 30px;
}
