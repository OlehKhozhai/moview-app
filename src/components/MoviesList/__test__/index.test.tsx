import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import MoviesList from '../index';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('MoviesList component', () => {
  test('should render MoviesList component by default', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <MoviesList isLoading={true} movies={[]} />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render MoviesList component with movies', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <MoviesList
          isLoading={false}
          movies={[
            {
              id: 1,
              budget: 100,
              genres: ['Documentary'],
              overview: 'Overview',
              poster_path: 'https://via.placeholder.com/330x450',
              title: 'Movie 1',
              release_date: '1999',
              revenue: 102,
              runtime: 12312,
              tagline: 'tagline',
              vote_average: 4,
              vote_count: 100,
            },
            {
              id: 2,
              budget: 100,
              genres: ['Documentary'],
              overview: 'Overview',
              poster_path: 'https://via.placeholder.com/330x450',
              title: 'Movie 1',
              release_date: '1999',
              revenue: 102,
              runtime: 12312,
              tagline: 'tagline',
              vote_average: 4,
              vote_count: 100,
            },
          ]}
        />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
