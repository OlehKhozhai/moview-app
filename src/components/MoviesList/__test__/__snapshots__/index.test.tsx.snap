// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList component should render MoviesList component by default 1`] = `
<div
  className="root"
>
  <h2>
    Loading
  </h2>
</div>
`;

exports[`MoviesList component should render MoviesList component with movies 1`] = `
<div
  className="root"
>
  <h3
    className="counter"
  >
    2
     movies found
  </h3>
  <li
    className="root"
    onClick={[Function]}
  >
    <div
      className="root dropdown"
    >
      <span
        className="threeDots"
        onClick={[Function]}
      >
        ...
      </span>
    </div>
    <img
      alt="Movie 1"
      className="image"
      height={495}
      src="https://via.placeholder.com/330x450"
      width={330}
    />
    <div
      className="titleAndImageWrapper"
    >
      <h4
        className="title"
      >
        Movie 1
      </h4>
      <span
        className="year"
      >
        1999
      </span>
    </div>
    <h5
      className="genre"
    >
      <span>
         
        Documentary
         |
      </span>
    </h5>
  </li>
  <li
    className="root"
    onClick={[Function]}
  >
    <div
      className="root dropdown"
    >
      <span
        className="threeDots"
        onClick={[Function]}
      >
        ...
      </span>
    </div>
    <img
      alt="Movie 1"
      className="image"
      height={495}
      src="https://via.placeholder.com/330x450"
      width={330}
    />
    <div
      className="titleAndImageWrapper"
    >
      <h4
        className="title"
      >
        Movie 1
      </h4>
      <span
        className="year"
      >
        1999
      </span>
    </div>
    <h5
      className="genre"
    >
      <span>
         
        Documentary
         |
      </span>
    </h5>
  </li>
</div>
`;
